pkg/registry/blobs.go

```diff
@@ -46,11 +51,26 @@ func isBlob(req *http.Request) bool {
 
 // blobs
 type blobs struct {
-	// Blobs are content addresses. we store them globally underneath their sha and make no distinctions per image.
-	contents map[string][]byte
 	// Each upload gets a unique id that writes occur to until finalized.
 	uploads map[string][]byte
 	lock    sync.Mutex
+
+	bh BlobHandler
+}
+
+// BlobHandler is the interface for the storage layer underneath this registry.
+type BlobHandler interface {
+	// Stat returns the size of the blob whose hash is specified,
+	// if it exists. If not, it returns (0, error).
+	Stat(repo name.Repository, h v1.Hash) (int64, error)
+
+	// Get returns true and a reader for consuming the blob specified with the hash,
+	// if it exists.  It now, it returns (nil, error).
+	Get(repo name.Repository, h v1.Hash) (io.ReadCloser, error)
```

Comment:
Something occurred to me about this API, thinking through how I could use it with kontain.me -- this API would require kontain.me to proxy and serve blobs from GCS, instead of HTTP redirecting like I do today, which is a lot faster and cheaper than proxying through Cloud Run.

This might just mean that `pkg/registry` isn't going to be a good fit for kontain.me, and that's fine, but I wanted to surface that here since I think it's a pretty nice use case that isn't supported by this proposal.